src/Sylius/Bundle/CoreBundle/Validator/Constraints/HasAllPricesDefinedValidator.php

```diff
@@ -22,6 +22,10 @@ final class HasAllPricesDefinedValidator extends ConstraintValidator
 {
     public function validate($value, Constraint $constraint): void
     {
+        if ($value->getProduct() === null) {
```

Comment:
IMO the assertion about the constraint should be the first check in this validator